src/uct/ib/dc/dc_mlx5_ep.h

```diff
@@ -231,7 +238,14 @@ static UCS_F_ALWAYS_INLINE ucs_status_t
 uct_dc_mlx5_ep_basic_init(uct_dc_mlx5_iface_t *iface, uct_dc_mlx5_ep_t *ep)
 {
     ucs_arbiter_group_init(&ep->arb_group);
-    ep->dci   = UCT_DC_MLX5_EP_NO_DCI;
+
+    if (uct_dc_mlx5_iface_is_dci_rand(iface)) {
+        /* coverity[dont_call] */
+        ep->dci = rand() % iface->tx.ndci;
```

Comment:
must NOT use rand() rand since it changes global rand state and can break applications/tests which check data validity
need to use rand_r()